app/controllers/contacts_controller.rb

```diff
@@ -3,13 +3,13 @@ class ContactUs::ContactsController < ApplicationController
   def create
     @contact = ContactUs::Contact.new(params[:contact_us_contact])
 
-    if verify_recaptcha(model: @contact) && @contact.save
-      redirect_to(ContactUs.success_redirect || '/', :notice => _('Contact email was successfully sent.'))
-    else
-      flash[:alert] = _('Captcha verification failed, please retry.')
-      redirect_to request.referrer
-      #render_new_page
+    unless user_signed_in?
+      unless verify_recaptcha(model: @contact) && @contact.save
```

Comment:
We should technically do the @contact.save check regardless of whether or not the user is signed in. Would also be good to do a render instead of redirect on failure so that the user doesn't lose the text they've entered in the fields.
Maybe something like (pseudocode .... you might need to tweak):
```ruby
@contact = ContactUs::Contact.new(params[:contact_us_contact])
flash[:alert] = nil

if !user_signed_in? && !verify_recaptcha(model: @contact)
  flash[:alert] = _('Captcha verification failed, please retry.')
end
if !flash[:alert].present? && @contact.save
  redirect_to(ContactUs.success_redirect || '/', :notice => _('Contact email was successfully sent.'))
else
  flash[:alert] = _('Unable to submit your request') unless flash[:alert].present?
  render 'new'
end
```